pkg/apis/pingcap/v1alpha1/tidbcluster.go

```diff
@@ -971,8 +971,9 @@ func (tc *TidbCluster) TiKVIsAvailable() bool {
 	return true
 }
 
-func (tc *TidbCluster) AllTiKVsAreAvailable() bool {
-	if len(tc.Status.TiKV.Stores) != int(tc.Spec.TiKV.Replicas) {
+func (tc *TidbCluster) AllTiKVsAreAvailable(tolerateSingleTiKVOutage bool) bool {
+	if (!tolerateSingleTiKVOutage && len(tc.Status.TiKV.Stores) != int(tc.Spec.TiKV.Replicas)) ||
+		(tolerateSingleTiKVOutage && int(tc.Spec.TiKV.Replicas-1) != len(tc.Status.TiKV.Stores)) {
```

Comment:
@BornChanger 
would this check return false if we enable `tolerateSingleTiKVOutage` and don't observe a WAL corruption (ie. replicas == stores)? should it be stores >= replicas - 1

Reply:
It's not a common fix but to work-around a single TiKV node crashloop. So, the process is that we observed one TiKV crashes due to the wal corruption issue, THEN turn on tolerateSingleTiKVOutage of the restore CR.